mir-module/src/main/resources/xsl/import/datacite2mods.xsl

```diff
@@ -85,10 +85,50 @@
         </xsl:if>
     </xsl:template>
 
+    <!-- Template for affiliation according to Datacite version 4.4  -->
     <xsl:template match="affiliation">
-        <mods:affiliation>
-            <xsl:value-of select="text()" />
-        </mods:affiliation>
+        <xsl:choose>
+            <xsl:when test="normalize-space(text()) and normalize-space(@affiliationIdentifierScheme)">
+                <mods:affiliation>
+                    <!-- Set valueURI if affiliationIdentifier exists and is not empty -->
+                    <xsl:if test="@affiliationIdentifier and normalize-space(@affiliationIdentifier)">
+                        <xsl:attribute name="valueURI">
+                            <xsl:value-of select="@affiliationIdentifier"/>
+                        </xsl:attribute>
+                    </xsl:if>
+
+                    <!-- If schemeURI exists and is not empty, use it as authorityURI -->
+                    <xsl:choose>
+                        <xsl:when test="normalize-space(@schemeURI)">
+                            <xsl:attribute name="authorityURI">
+                                <xsl:value-of select="@schemeURI"/>
+                            </xsl:attribute>
+                        </xsl:when>
+                        <!-- If schemeURI is missing or empty, use predefined values for ROR, ISNI, GRID -->
+                        <xsl:when test="not(normalize-space(@schemeURI))">
+                            <xsl:choose>
+                                <xsl:when test="@affiliationIdentifierScheme = 'ROR'">
+                                    <xsl:attribute name="authorityURI">https://ror.org/</xsl:attribute>
+                                </xsl:when>
+                                <xsl:when test="@affiliationIdentifierScheme = 'ISNI'">
+                                    <xsl:attribute name="authorityURI">https://isni.org/</xsl:attribute>
+                                </xsl:when>
+                                <xsl:when test="@affiliationIdentifierScheme = 'GRID'">
+                                    <xsl:attribute name="authorityURI">https://grid.ac/</xsl:attribute>
+                                </xsl:when>
```

Comment:
not sure if no authorityURI is specified in the context whether it is valid?

Reply:
as of [mods:affiliation](https://www.loc.gov/standards/mods/userguide/name.html#affiliation) I don't see a problem with that